com.unity.ml-agents/Runtime/Policies/HeuristicPolicy.cs

```diff
@@ -31,7 +34,9 @@ public void RequestDecision(AgentInfo info, List<ISensor> sensors)
             StepSensors(sensors);
             if (!info.done)
             {
-                m_LastDecision = m_Heuristic.Invoke();
+                // Reset m_LastDecision each time.
+                 m_LastDecision = new float[m_numActions];
```

Comment:
I couldn't decide whether or not to reuse the array here. I figure this approach is safer, and this isn't performance-critical enough to worry about the allocation/GC

Reply:
I think it depends on the number of agents and the size of the action array.  If there is a large action space and a QA person is trying to make a demo file to use in training but they are hitting GC hitches every few seconds that could make it 1. hard to actually make a good demo. 2. frustrate the hell out of that person.

Reply:
Discussed offline - we need to do a deeper review of array sharing and/or copying around this (and the other policies). Logged as https://jira.unity3d.com/browse/MLA-892 for followup.